### ISSUE TITLE
Migrate odata connector from forked camel-olingo to fuse dependencies

### ISSUE THREAD

**Issue by phantomjinx** [syndesisio/syndesis#4693]
- Check the newly-built camel-olingo4 dependencies to ensure they have the same code/fixes as the forked version inside the odata-connector.
- If so, remove the forked source code and update the odata-connector pom.xml accordingly.

**Comment by phantomjinx** [comment]
Blocked by -> [PR-2789](https://github.com/apache/camel/pull/2789): needs merging and new camel build to come around.

**Comment by phantomjinx** [comment]
Summary of camel fixes needed in new camel-olingo4 library build:
- [CAMEL-13082](https://issues.apache.org/jira/browse/CAMEL-13082)
- [CAMEL-13185](https://issues.apache.org/jira/browse/CAMEL-13185)

FYI @oscerd HTH

**Comment by oscerd** [comment]
You can create the PRs on the jboss-fuse/camel project

**Comment by phantomjinx** [comment]
@oscerd 
Is that a private repository as I cannot seem to find it?

**Comment by oscerd** [comment]
Yes, it is. I'll add the commit for you downstream, in the mean time @janstey can you add @phantomjinx to the fuse organization? thanks

**Comment by phantomjinx** [comment]
@oscerd 
Just to update.
camel-olingo4-2.21.0.fuse-740001 was brought around on 9/4/2019. It contains all the changes mentioned in the previous comments. Unfortunately, CAMEL-13316 was found subsequently.
The fix has been merged into the camel repo's master branch on ~21/3/2019 yet does not appear in the library.

At some point, you'll bump to another version of the fork. However, I am continuing work on the odata connector and I know there'll be at least 1 other issue to fix, ie. [#5067](https://github.com/syndesisio/syndesis/issues/5067). There doesn't seem much point in migrating to the dependent library just yet. Any thoughts on a strategy going forward as I don't want to give the impression that we have our own camel-olingo4 and developing it here (see [here](https://issues.jboss.org/browse/ENTESB-10119)), after all this is my current operating methodology:

1. Find issue and fix in syndesis code
2. Take patch from syndesis code to file
3. Apply patch to local camel repo and push to my fork
4. Create new issue on camel bugs list for olingo4 component
5. Create PR of change and link to it in the camel issue 
6. Wait for new fuse library to come around and do a diff from syndesis code to source jar, checking all fixes are present

Thoughts welcome.

cc @heiko-braun

**Comment by oscerd** [comment]
You need to raise an ENTESB JIRA to backport what is missing on the component. In my opinion the camel code cannot be duplicated in two different projects, because the sustaining effort is duplicated in this way. I would include the fix for CAMEL-13116 to Fuse Camel, release a new camel dev build, remove the code from Syndesis and use the camel-olingo4 dependency, for #5067 you can always fix it in the Fuse Camel version and upstream.

**Comment by phantomjinx** [comment]
Hi @janstey,
Can you add me to the jboss-fuse organization so I can see jboss-fuse/camel please?
Thanks!

**Comment by phantomjinx** [comment]
Created [ENTESB-10479](https://issues.jboss.org/browse/ENTESB-10479) requesting backport of [CAMEL-13316](https://issues.apache.org/jira/browse/CAMEL-13316).

@oscerd 
Assuming I get access to jboss-fuse, are instructions available for creating dev release build (presume mvn deploy)?

Thanks.

**Comment by heiko-braun** [comment]
@phantomjinx I have send you an invitation
